### PR TITLE
Add Voron Trident profiles, bed meshes, variants

### PR DESCRIPTION
**Pull request by RagingRoosevelt**
Adds a 250,300,350 Voron Trident printer profiles to Cura.  Adds toolhead variants matching existing VoronDesign printer profiles.

The profile is based on existing Voron2_Base printer profile.

The bed STL is courteously provided with permission from @hartk1213 ([via his github](https://github.com/hartk1213/MISC/tree/f5b1685f1d56b7e2ab853a98741ae614131b116c/Voron%20Mods/SuperSlicer/Bed_Models/Trident)).
image

Please let me know if this requires any modification. I am happy to make changes.

### CODE REVIEW

**Comment by hartk1213** [comment]
> Adds a 250,300,350 Voron Trident printer profiles to Cura.  Adds toolhead variants matching existing VoronDesign printer profiles.
> 
> The profile is based on existing Voron2_Base printer profile.
> 
> The bed STL is courteously provided with permission from @hartk1213 ([via his github](https://github.com/hartk1213/MISC/tree/f5b1685f1d56b7e2ab853a98741ae614131b116c/Voron%20Mods/SuperSlicer/Bed_Models/Trident)).
> image
> 
> 
> Please let me know if this requires any modification. I am happy to make changes.
> 

Awesome glad those will work out👍

**Comment by jellespijker** [comment]
Hi @hartk1213 thank you for contributing those printers, please take not of the failing unit tests. A quick glance showed that the file size of the bed meshes is to big

**Comment by RagingRoosevelt** [comment]
> bed meshes is to big

I'll try to get the filesize reduced on those.  I don't want to burden hartk with the filesize reduction with that when they were already gracious enough to allow use of the files.

Separate question:

Do I need to address the `Overriding {{ value }} with the same value (default_value: {{ value }}) as defined in parent definition` issue prior to acceptance, or is that OK?  The base printer here is from the same manufacturer but there are no guarantees that it will remain the same base size.

**Comment by jellespijker** [comment]
The `Overriding {{ value }}...` can be ignored they are after all suggestions. But please motivate the reason why, such that it is a concise decision. The `machine-*` settings are probably very machine specific, you can leave these be.

File: resources/definitions/voron_trident_350.def.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+        "platform": "voron_trident_350mm_bed_model.stl",
+        "quality_definition": "voron2_base",
+        "author": "RagingRoosevelt, hartk"
+    },
+    "overrides":
+    {
+        "machine_name":         { "default_value": "VORON Trident 350" },
+        "machine_width":        { "default_value": 350 },
+        "machine_depth":        { "default_value": 350 },
+        "machine_height":       { "default_value": 250 }
```
[the remark this answers is not in the document]
Comment (RagingRoosevelt) [reply resources/definitions/voron_trident_350.def.json]: Adjusted def to remove this repetition

File: resources/definitions/voron_trident_300.def.json [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+        "platform": "voron_trident_300mm_bed_model.stl",
+        "quality_definition": "voron2_base",
+        "author": "RagingRoosevelt, hartk"
+    },
+    "overrides":
+    {
+        "machine_name":         { "default_value": "VORON Trident 300" },
+        "machine_width":        { "default_value": 300 },
+        "machine_depth":        { "default_value": 300 },
+        "machine_height":       { "default_value": 250 }
```
[the remark this answers is not in the document]
Comment (RagingRoosevelt) [reply resources/definitions/voron_trident_300.def.json]: Adjusted def to remove this repetition

**Comment by RagingRoosevelt** [comment]
Since the Voron Trident is a different model from the Voron 2.x, inheriting dimensions doesn't make sense.  

I've updated the inheritance patterns to
```
voron2_base/
├─ voron2_250
├─ voron2_300
├─ voron2_350
├─ voron0_120
├─ voron_trident_base/
   ├─ voron_trident_250
   ├─ voron_trident_300
   ├─ voron_trident_350
```

so that repeated definitions only occur in once place.  I believe this addresses the spirit of those guidelines in addition to respecting the realities of the way Voron models are split up.

---

Ideally, I think the structure would be
```
voron_base/
├─ voron2_base/
│  ├─ voron2_250
│  ├─ voron2_300
│  ├─ voron2_350
├─ voron_0_base/
│  ├─ voron0_120
├─ voron_trident_base/
│  ├─ voron_trident_250
│  ├─ voron_trident_300
│  ├─ voron_trident_350
⋮
```
but this is out-of-scope for this PR.  If this makes sense, I'd be happy to work on another PR to tackle this, though.

**Review by Joeydelarago — APPROVED** [review]

**Comment by Joeydelarago** [comment]
Looking Good 👍 Thanks for the cntribution!

**Pull request merged**